central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -84,6 +85,12 @@ func (ds *datastoreImpl) UpsertScanConfiguration(ctx context.Context, scanConfig
 	return ds.storage.Upsert(ctx, scanConfig)
 }
 
+// UpsertNoLockScanConfiguration upserts scan config like UpsertScanConfiguration but does not create a lock
+func (ds *datastoreImpl) UpsertNoLockScanConfiguration(ctx context.Context, scanConfig *storage.ComplianceOperatorScanConfigurationV2) error {
```

Comment:
So to follow our typical pattern this should be `upsertNoLockScanConfiguration`.

Then the last 2 lines in `UpsertScanConfiguration` should be removed and it should call `upsertNoLockScanConfiguration` instead.